### PR TITLE
TheMediaGrid: Added processing of imp[].ext.data

### PR DESCRIPTION
**Pull request by TheMediaGrid**
Added processing of imp[].ext.data to add imp[].ext.gpid = imp[].ext.data.adserver.adslot

### CODE REVIEW

File: adapters/grid/grid.go [hunk 1]
```diff
@@ -37,6 +54,21 @@ func processImp(imp *openrtb2.Imp) error {
...
 
+func setImpExtData(imp openrtb2.Imp) openrtb2.Imp {
+	var ext ExtImp
+	if err := json.Unmarshal(imp.Ext, &ext); err != nil {
+		return imp
+	}
+	if ext.Data != nil && ext.Data.AdServer != nil && ext.Data.AdServer.AdSlot != "" {
+		ext.Gpid = ext.Data.AdServer.AdSlot
+	}
+	extJSON, err := json.Marshal(ext)
```
Comment (hhhjort) [inline adapters/grid/grid.go:65]: You should put this marshalling up in the above if clause. If you have made no changes to the ext data, there is no reason to remarshal it that I can see.

Comment (TheMediaGrid) [reply adapters/grid/grid.go]: Done

**Review by hhhjort — APPROVED** [review]
LGTM

**Comment by TheMediaGrid** [comment]
@bsardo Hello, are there any questions that block the merge?

**Review by bsardo — APPROVED** [review]
LGTM

**Pull request merged**